pybamm/models/submodels/base_submodel.py

```diff
@@ -135,6 +135,21 @@ def domain(self, domain):
     def domain_Domain(self):
         return self._domain, self._Domain
 
+    def get_parameter_info(self, by_submodel=False):
+        """
+        Extracts the parameter information and returns it as a dictionary.
+        To get a list of all parameter-like objects without extra information,
+        use :py:attr:`model.parameters`.
+
+        Returns
+        -------
+        NotImplementedError:
+            This method is not available for direct use on submodels. It is recommended to use on the full model.
```

Comment:
```suggestion
            This method is not available for direct use on submodels since the submodel may contain coupled variables that depend on other submodels, in which case some parameters may be missed. It is recommended to use on the full model.
```